service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -31,42 +34,39 @@
     @PostMapping("/about-to-submit")
     public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
-        CaseData caseData = getCaseData(caseDetails);
-        Object migrationId = getMigrationId(caseDetails);
-        String familyManCaseNumber = caseData.getFamilyManCaseNumber();
+        Object migrationId = caseDetails.getData().get(MIGRATION_ID_KEY);
 
-        if (isCorrectCase(migrationId, familyManCaseNumber)) {
-            log.info("Removing hearings from case reference {}", caseDetails.getId());
-            caseDetails.getData().put("hearingDetails", removeHearings(caseData.getHearingDetails()));
-            caseDetails.getData().remove(MIGRATION_ID_KEY);
+        if ("FPLA-2469".equals(migrationId)) {
+            run2469(caseDetails);
         }
 
+        caseDetails.getData().remove(MIGRATION_ID_KEY);
         return respond(caseDetails);
     }
 
-    private boolean isCorrectCase(Object migrationId, String familyManCaseNumber) {
-        return "FPLA-2437".equals(migrationId) && "ZW20C50003".equals(familyManCaseNumber);
-    }
 
-    private Object getMigrationId(CaseDetails caseDetails) {
-        return caseDetails.getData().get(MIGRATION_ID_KEY);
-    }
+    private void run2469(CaseDetails caseDetails) {
+        CaseData caseData = getCaseData(caseDetails);
 
-    private List<Element<HearingBooking>> removeHearings(List<Element<HearingBooking>> hearings) {
-        for (int i = 0; i < 3; i++) {
-            assertHearingDate(hearings.get(i).getValue());
-            log.info("hearing {} has correct date", i);
-        }
-        log.info("Removing hearing 3");
-        hearings.remove(2);
-        log.info("Removing hearing 1");
-        hearings.remove(0);
-        return hearings;
-    }
+        if ("SN20C50010".equals(caseData.getFamilyManCaseNumber())) {
+            List<Element<HearingBooking>> hearings = caseData.getHearingDetails();
+
+            int hearingIndex = 0;
+            LocalDate expectedHearingDate = LocalDate.of(2020, Month.OCTOBER, 14);
+
+            if (hearings.size() < hearingIndex + 1) {
+                throw new IllegalArgumentException(format("Case %s has %s hearing(s), expected at least %s ",
+                    caseDetails.getId(), hearings.size(), hearingIndex + 1));
+            }
+
+            if (!expectedHearingDate.equals(hearings.get(hearingIndex).getValue().getStartDate().toLocalDate())) {
```

Comment:
vv minor: I think it would be nice to extract `hearings.get(hearingIndex).getValue().getStartDate().toLocalDate())` to a variable